instance/instance.go

```diff
@@ -260,6 +272,32 @@ func (i *Instance) Prefix() string {
 	return i.Domain + "/"
 }
 
+// RegisterPassword replace the instance registerToken by a password
+func (i *Instance) RegisterPassword(pass string, tok string) error {
+	if i.RegisterToken == "" {
+		return errors.New("Missing registerToken")
+	}
+	if i.RegisterToken != tok {
```

Comment:
This should be constant time equality check (to avoid timing attacks): see [subtle package](https://golang.org/pkg/crypto/subtle/#ConstantTimeCompare)